File: jsimple-io/src/main/java/jsimple/io/IOException.java
/*
 * Copyright (c) 2012-2015 Microsoft Mobile.  All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 *
 * This file is based on or incorporates material from Apache Harmony
 * http://harmony.apache.org (collectively, "Third Party Code"). Microsoft Mobile
 * is not the original author of the Third Party Code. The original copyright
 * notice and the license, under which Microsoft Mobile received such Third Party
 * Code, are set forth below.
 *
 *
 * Copyright 2006, 2010 The Apache Software Foundation.
 *
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jsimple.io;

import jsimple.util.BasicException;

/**
 * This is a platform independent class used to represent I/O exceptions, taking the place of java.io.IOException in
 * JSimple based code.
 *
 * @author Bret Johnson
 * @since 10/6/12 5:27 PM
 */
public class IOException extends BasicException {
    public IOException(Throwable cause) {
        super(cause);
    }

    public IOException(String message) {
        super(message);
    }

    public IOException(String message, Object arg1) {
        super(message, arg1);
    }

    public IOException(String message, Object arg1, Object arg2) {
        super(message, arg1, arg2);
    }

    public IOException(String message, Object... args) {
        super(message, args);
    }
}

File: jsimple-io/c#/jsimple/io/IOException.cs
using System;

namespace jsimple.io
{

	using BasicException = jsimple.util.BasicException;

	/// <summary>
	/// This is a platform independent class used to represent I/O exceptions, taking the place of java.io.IOException in
	/// JSimple based code.
	/// 
	/// @author Bret Johnson
	/// @since 10/6/12 5:27 PM
	/// </summary>
	public class IOException : BasicException
	{
		public IOException(Exception cause) : base(cause)
		{
		}

		public IOException(string message) : base(message)
		{
		}

		public IOException(string message, object arg1) : base(message, arg1)
		{
		}

		public IOException(string message, object arg1, object arg2) : base(message, arg1, arg2)
		{
		}

		public IOException(string message, params object[] args) : base(message, args)
		{
		}
	}

}